### DISCUSSION TITLE
Optimizing the renderer

### DISCUSSION THREAD

**Discussion by henke37** [airsdk/Adobe-Runtime-Support#2181] [General]
The flash player is generally a great and efficient rendering engine. But it has its weak spots. Are there any plans on working on optimizing these spots?

**Comment by ajwfrost** [comment]
If there are particular, repeatable use cases where the rendering is particularly slow, we can take a look into it, but there are often a number of trade-offs or optimisations that can be made to content to work better with the engine..  plus of course, rendering can happen using CPU (normal/auto and direct mode) or with the GPU (which translates to different APIs depending on what platform you're running on) so it's not always immediately clear what code is actually needing the improvement!

So we're not pro-actively looking into the rendering routines, they've been relatively untouched for many years now (with SIMD acceleration and multithreading support). Let us know if you have a simple app that can demonstrate a problem area though!

thanks

**Comment by henke37** [comment]
I dunno about "app", but I do have a very expensive movie I found ages ago. When I run it in fullscreen _at least_ every two of three frames are skipped.

Attached is an instrumented version that I prepared. It measures just how bad the frame skipping is.

[futureTest.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/9706306/futureTest.zip)
